t/re/reg_mesg.t

```diff
@@ -220,6 +220,8 @@ my @death =
  '/\g/' => 'Unterminated \g... pattern {#} m/\g{#}/',
  '/\g{1/' => 'Unterminated \g{...} pattern {#} m/\g{1{#}/',
  '/\g{-abc}/' => 'Group name must start with a non-digit word character {#} m/\g{-{#}abc}/',
+ '/\g{1-1}/'  => 'Sequence \g{... not terminated {#} m/\g{1{#}-1}/',
```

Comment:
To a human, it is obvious that this is terminated; just syntactically invalid.  The parser can be forgiven for not being able to detect that.  But it is unfortunate that the existing diagnostic doesn't give that as a possibility.  You have to go to perldiag to find that a syntax error is a possibility,  We likely can't change existing diagnostics, but this is a new error, and so a clearer error message could be devised.

Reply:
I was following the example of named backreferences:
```
$ perl -e '/\g{foo-bar}/'
Sequence \g{... not terminated in regex; marked by <-- HERE in m/\g{foo <-- HERE -bar}/ at -e line 1.
```